test/Microsoft.TestPlatform.TestUtilities/IntegrationTestBase.cs

```diff
@@ -727,7 +727,15 @@ protected string BuildMultipleAssemblyPath(params string[] assetNames)
 
         protected static string GetResultsDirectory()
         {
-            var directoryPath = Path.Combine(Path.GetTempPath(), Guid.NewGuid().ToString());
+            // DUMP_TEMP_PATH will be used as temporary storage location 
+            // for the dumps, this won't affect the dump uploads. Just the place where 
+            // we store them before moving them to the final folder. 
+
+            // AGENT_TEMPDIRECTORY is AzureDevops variable, which is set to path 
+            // that is cleaned up after every job. This is preferable to use over 
+            // just the normal temp. 
+            var temp = Environment.GetEnvironmentVariable("VSTEST_DUMP_TEMP_PATH") ?? Environment.GetEnvironmentVariable("AGENT_TEMPDIRECTORY") ?? Path.GetTempPath();
```

Comment:
This should use only agent_temp or %temp%. Dump temp should be used only for process dumps.

Reply:
Copy-paste accident :)